src/app/modules/issue/issue.service.ts

```diff
@@ -0,0 +1,182 @@
+    return errAsync(new FormNotFoundError())
+  }
+  return ResultAsync.fromPromise(
+    FormIssueModel.create({
+      formId: formId,
+      issue: issue,
+      ...(email && { email: email }),
+    }),
+    (error) => {
+      logger.error({
+        message: 'Database error when creating form issue document',
+        meta: {
+          action: 'insertFormIssue',
+          formId,
+        },
+        error,
+      })
+
+      return new DatabaseError('Form issue could not be created')
+    },
+  )
+}
+
+/**
+ * Checks if there are any prior issues reported today given a formId.
+ * The `_id` field of the form is used to check for prior issues, even though
+ * mongo objectId is accurate only up to seconds, it should be fine, as we do
+ * not expect more than 1 issue complain per form per second. `created` field is
+ * not used for now as the column is not being indexed (to reduce additional
+ * write overhead).
+ * In the unlikely chance of this happening, form admin and collaborators may
+ * receive up to the number of issues reported per second per form.
+ * @param formIssue the formIssue object that was just inserted
+ * @returns ok(true) if successfully inserted
+ * @returns err(DatabaseError) on database error
+ */
+const getIsFirstIssueForFormToday = ({
+  formIssue,
+}: {
+  formIssue: IFormIssueSchema
+}): ResultAsync<boolean, DatabaseError> => {
+  const startOfDay =
+    formIssue._id instanceof ObjectId
+      ? getStartOfDay({ date: formIssue._id.getTimestamp() })
+      : getStartOfDay()
+
+  const issueId = formIssue._id as ObjectId
+  const queryFilter = {
+    formId: formIssue.formId,
+    _id: {
+      $gte: ObjectId.createFromTime(startOfDay.getTime() / 1000),
+      $lt: issueId,
+    },
+  }
+  return ResultAsync.fromPromise(
+    FormIssueModel.countDocuments(queryFilter).limit(1).exec(),
+    (error) => {
+      logger.error({
+        message: 'Database error when creating form issue document',
+        meta: {
+          formId: formIssue.formId,
+          action: 'insertFormIssue',
+        },
+        error,
+      })
+      return new DatabaseError(
+        'Failed to count the document in the formIssue collection',
+      )
+    },
+  ).andThen((count) => {
+    return okAsync(count == 0)
+  })
+}
+
+export const getIsFirstIssueForFormTodayForTesting = getIsFirstIssueForFormToday
+
+/**
+ * Notifies form admin and collaborators whenever public reports an issue for a
+ * form first time of any day.
+ * Since the current logic will only send email at most once a day per form, if
+ * the email sending fails for whatever reason, we will only print a log for
+ * now, and will not throw the error to the caller.
+ * @param form the form object where public reports an issue
+ * @param formIssue the form issue object where the issue is recorded
+ */
+export const notifyFormAdmin = ({
+  form,
+  formIssue,
+}: {
+  form: IPopulatedForm
+  formIssue: IFormIssueSchema
+}): Promise<boolean> => {
```

Comment:
Hm. Any specific requirement to not return `ResultAsync`?

Reply:
Because the error needs to be silenced, if I use ResultAsync.mapErr(), it will translate <T, E> to <T, F>, but the error will still be propagated, and i didn't want this to happen. let me know if you have an alternative.